### DISCUSSION TITLE
Digital selective calling

### DISCUSSION THREAD

**Discussion by Taxom** [jvde-github/AIS-catcher#63] [Ideas]
The idea of expanding functionality, add decoding DSC GMDSS (156.525 MHz, Marine VHF radio Channel 70) with their binding to the sender.

**Comment by jvde-github** [comment]
I see that I still have it in the TO DO list and looked at it and prototyped something in Python a while ago to decode but do you see a lot of interesting traffic on this channel? I read somewhere they were going to phase this out on the FM band.

**Reply by jimarndt** [reply]
I did some monitoring of VHF DSC (Channel 70) in the Los Angeles/Long Beach area with a friend, and honestly the amount of traffic was surprisingly low. It seems that, at least in Southern California, the recreational marine community doesn't make much use of DSC calling, position polling, or position reports. Even DSC test calls appear to be relatively infrequent.

That said, I still think DSC decoding would be a valuable addition to AIS-Catcher. Distress alerts, position requests/replies, routine calls, safety calls, and test calls are all interesting information. Messages containing position information, such as Distress Alerts and Position Replies, could even be displayed directly on the map alongside AIS targets, while routine calls, safety calls, and test calls could appear in an event log.

Another interesting application is that DSC Position Requests and Position Replies can be used by groups of boats to privately exchange positions without announcing them over voice. It's not encrypted, but today very few people are monitoring and decoding DSC traffic, so it provides a level of privacy that voice communications do not.

HF DSC, on the other hand, is very active. There is regular coast station traffic, commercial vessel traffic, safety messages, and DSC test calls. The amateur radio community has also taken advantage of this by monitoring the five standard HF DSC frequencies (2187.5, 4207.5, 6312, 8414.5, and 12577 kHz) to observe HF propagation conditions.

There are already communities built around this, such as the DSC Group on Groups.io and YADDNet (https://www.yaddnet.org/), which collects DSC packets from receiving stations worldwide. While it accepts both VHF and HF DSC, there are currently very few VHF monitoring stations contributing data.

A commercial captain friend of mine actually uses YADDNet to verify his weekly HF DSC test calls. He often performs the required test while tied up at the dock, where the high RF noise environment makes HF reception much more difficult than when offshore. As a result, he doesn't always receive the USCG acknowledgment from San Francisco. However, YADDNet frequently shows that multiple shore stations successfully received his transmission. He prints those reception reports and keeps them as documentation for his annual USCG inspection, demonstrating that he is performing the required weekly DSC test calls.

I think this could represent an interesting new application for the AIS-Catcher/Daisy ecosystem. In addition to decoding DSC traffic, a distributed network of Daisy receivers could become a valuable source of worldwide HF propagation data, much like the AIS-Catcher network has done for AIS. It could also significantly increase the number of VHF DSC monitoring stations contributing data to projects like YADDNet.

Out of curiosity, is Daisy's RF front end capable of tuning down to the standard HF DSC frequencies, or is it limited by the front-end filtering? Since Daisy has a single SMA antenna input, I wasn't sure whether the existing hardware could support both an HF and a VHF antenna (perhaps using a diplexer), or whether HF support would require hardware changes. If the receiver is already broadband, this seems like it could be an interesting future application.

**Reply by astuder** [reply]
Thank you for the detailed post. I learned a lot :)

2-12 MHz is too far away from marine VHF (~156-162 MHz) to be usefully covered by the same receiver. AIS receivers typically have a band or high-pass filter on the front end. In addition, any impedance matching would be completely off at those frequencies. 

VHF DSC is within the marine band, so that seems feasible on a technical level.

**Reply by jimarndt** [reply]
Thanks for the explanation—that makes perfect sense. I hadn't considered the front-end filtering and impedance mismatch across such a wide frequency range, so I appreciate the clarification.

I'm going to keep the dAISy-catcher dedicated to dual-channel AIS, and instead add a second RTL-SDR to monitor VHF DSC on Channel 70. The long-term goal is a Raspberry Pi-based shore station overlooking the Southern California coast that receives both AIS and DSC simultaneously from a single VHF antenna (using a passive splitter), with both data streams feeding a common dashboard.

I'm curious if anyone here has experience with VHF DSC decoding on Raspberry Pi (Debian ARM64).

So far I've found:

Intercept
blackwellj/dsc-decoder (Docker container)

Both appear promising, but I haven't found much information from people actually running them on a Raspberry Pi.

Can anyone recommend a DSC decoder that runs well on Raspberry Pi, or share any lessons learned? Ultimately I'd like to decode Channel 70 directly from an RTL-SDR and have the decoder output JSON or another machine-readable format for integration with the AIS data.

Thanks again for pointing me in the right direction.

**Reply by astuder** [reply]
No DSC decoding, but this project monitoring VHF voice traffic seems somewhat related to your goals:
https://seattleboatradio.com/
https://github.com/boscacci/vhf-seattle

I think it uses Docker on the Pi, but the whole IT infrastructure thing is way above my pay grade :)

**Reply by jimarndt** [reply]
Thanks again for the pointers. I wanted to share a quick update in case it's helpful to others.

I was able to get the blackwellj/dsc-decoder project running natively on a Raspberry Pi 3 (Debian 13 ARM64) without Docker. The decoder itself is pure Python (NumPy/SciPy), so the x86-only Docker image isn't actually a limitation.

At the moment I'm debugging the SDR audio path. The RTL-SDR is definitely receiving Channel 70 transmissions from my Standard Horizon HX870/HX890 radios, but the decoder isn't yet detecting the DSC sync pattern. So the remaining issue appears to be the interface between the SDR/FM demodulator and the decoder, not ARM compatibility.

I'll post another update once I have it successfully decoding live DSC traffic.

**Reply by jimarndt** [reply]
> No DSC decoding, but this project monitoring VHF voice traffic seems somewhat related to your goals: https://seattleboatradio.com/ https://github.com/boscacci/vhf-seattle
> 
> I think it uses Docker on the Pi, but the whole IT infrastructure thing is way above my pay grade :)

Thanks for sharing that project—it was a fascinating rabbit hole. I spent quite a bit of time looking through it, and I'm really impressed with what was built. The automatic recording, transcription, channel logging, and AIS-based vessel/location correlation are all very clever. It definitely gave me a lot of ideas for where I'd eventually like to take my own project.

My immediate goal is still decoding AIS and DSC on the Pi, but once that's working I could definitely see adding VHF voice logging and transcription as another data source. Thanks again for pointing me to it!
